src/Hmac/Adapter/SymfonyRequestAdapter.php

```diff
@@ -0,0 +1,52 @@
+<?php declare(strict_types=1);
+
+namespace Starlit\Request\Authenticator\Hmac\Adapter;
+
+use Symfony\Component\HttpFoundation\Request;
+
+class SymfonyRequestAdapter implements RequestAdapterInterface
+{
+    /**
+     * @var Request
+     */
+    private $request;
+
+    public function __construct(Request $request)
+    {
+        $this->request = $request;
+    }
+
+    public function getMethod(): string
+    {
+        return $this->request->getMethod();
+    }
+
+    /**
+     * helper method to build a uri without reordering url parameters in a query string alphabetically
+     * which the interal Request::getUri function does (called normalize)
+     *
+     * @see SymfonyRequest::getUri()
+     * @see SymfonyRequest::getQueryString()
+     * @see SymfonyRequest::normalizeQueryString()
+     */
+    public function getUri(): string
+    {
+        $qs = $this->request->server->get('QUERY_STRING');
```

Comment:
Maybe rename `$qs` to `$queryString` instead in order to make it's use clearer?

Reply:
yes, just copied the method body from the symfony method, will change it.